### PR TITLE
Fix S2068 FN: Support utf-8 string literals

### LINKED ISSUE #6394: Fix S2068 FN: Support utf-8 string literals
### Description

FN for S2068 (DoNotHardcodeCredentials).

Using C#11 utf-8 string literals, the rule is not raising an issue.

### Repro steps

```csharp
void Utf8StringLiterals()
{
    var DBConnectionString1 = "Server=localhost; Database=Test; User=SA; Password=Secret123"u8; // FN
    var DBConnectionString2 = """Server=localhost; Database=Test; User=SA; Password=Secret123"""u8; // FN
    var DBConnectionString3 = """
        Server=localhost; Database=Test; User=SA; Password=Secret123
        """u8; // FN
}
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.

### PR DESCRIPTION
**Pull request by mary-georgiou-sonarsource**
Fixes #6394

### CODE REVIEW

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]
I think the GetStringValue override is not the right solution. It should be possible to do it one layer up in the inheritance chain.

File: analyzers/src/SonarAnalyzer.CSharp/Helpers/CsharpStringValueHelper.cs [hunk 1]
```diff
@@ -29,5 +31,18 @@ public class CsharpStringValueHelper : StringValueHelper<SyntaxKind, Interpolate
         protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
 
         protected override SyntaxToken Token(LiteralExpressionSyntax literalExpression) => literalExpression.Token;
+
+        public override string GetStringValue(SyntaxNode node, SemanticModel semanticModel)
+        {
+            if (base.GetStringValue(node, semanticModel) is { } stringValue)
```
[the remark this answers is not in the document]
Comment (mary-georgiou-sonarsource) [reply analyzers/src/SonarAnalyzer.CSharp/Helpers/CsharpStringValueHelper.cs]: Merged!

**Comment by mary-georgiou-sonarsource** [comment]
@martin-strecker-sonarsource please re-assign to @csaba-sagi-sonarsource after you review as I'll be away. 
We just synced on the matter.

**Review by martin-strecker-sonarsource — DISMISSED** [review]
I'm taking the implementation over from Mary.

**Review by cristian-ambrosini-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**